node/pkg/p2p/cutover.go

```diff
@@ -0,0 +1,131 @@
+		go func() {
+			time.Sleep(delay)
+			logger.Info("time to cut over to new quic-v1", zap.String("cutOverTime", cutOverTimeStr), zap.String("component", "p2pco"))
+			panic("p2pco: time to cut over to new quic-v1")
+		}()
+	}
+
+	return nil
+}
+
+// evaluateCutOverImpl performs the actual cut over check. It is a separate function for testing purposes.
+func evaluateCutOverImpl(logger *zap.Logger, cutOverTimeStr string, now time.Time) (bool, time.Duration, error) {
+	if cutOverTimeStr == "" {
+		return false, 0, nil
+	}
+
+	cutOverTime, err := time.Parse(cutOverFmtStr, cutOverTimeStr)
+	if err != nil {
+		return false, 0, fmt.Errorf(`failed to parse cut over time: %w`, err)
+	}
+
+	if cutOverTime.Before(now) {
+		logger.Info("cut over time has passed, should use new quic-v1", zap.String("cutOverTime", cutOverTime.Format(cutOverFmtStr)), zap.String("now", now.Format(cutOverFmtStr)), zap.String("component", "p2pco"))
+		return true, 0, nil
+	}
+
+	// If we get here, we need to wait for the cutover and then force a restart.
+	delay := cutOverTime.Sub(now)
+	logger.Info("still waiting for cut over time",
+		zap.Stringer("cutOverTime", cutOverTime),
+		zap.String("now", now.Format(cutOverFmtStr)),
+		zap.Stringer("delay", delay),
+		zap.String("component", "p2pco"))
+
+	return false, delay, nil
+}
+
+// getCutOverTimeStr returns the cut over time string based on the network ID passed in.
+func getCutOverTimeStr(networkID string) string {
+	if strings.Contains(networkID, "/mainnet/") {
+		return mainnetCutOverTimeStr
+	}
+	if strings.Contains(networkID, "/testnet/") {
+		return testnetCutOverTimeStr
+	}
+	return devnetCutOverTimeStr
+}
+
+// cutOverBootstrapPeers checks to see if we are supposed to cut over, and if so updates the bootstrap peers. It assumes that the string has previously been validated.
+func cutOverBootstrapPeers(bootstrapPeers string) string {
+	if shouldCutOver() {
+		bootstrapPeers = strings.ReplaceAll(bootstrapPeers, "/quic/", "/quic-v1/")
+	}
+
+	return bootstrapPeers
+}
+
+// cutOverAddressPattern checks to see if we are supposed to cut over, and if so updates the address patterns. It assumes that the string is valid.
+func cutOverAddressPattern(pattern string) string {
+	if shouldCutOver() {
+		if !strings.Contains(pattern, "/quic-v1") {
+			// These patterns are hardcoded so we are not worried about invalid values.
+			pattern = strings.ReplaceAll(pattern, "/quic", "/quic-v1")
+		}
+	}
+
+	return pattern
+}
+
+// validateBootstrapPeers ensures that the quic parameters in a bootstrap peers string are consistent. It is a separate function for testing purposes.
+func validateBootstrapPeers(bootstrapPeers string) error {
+	if !strings.Contains(bootstrapPeers, "/quic/") && !strings.Contains(bootstrapPeers, "/quic-v1/") {
+		return fmt.Errorf(`unexpected format, does not contain "/quic/" or  "/quic-v1/"`)
+	}
+
+	if strings.Contains(bootstrapPeers, "/quic/") && strings.Contains(bootstrapPeers, "/quic-v1/") {
```

Comment:
why does this matter? `cutOverBootstrapPeers` can safely handle either of these cases

Reply:
Removed this function.